reference/constraints/_payload-option.rst.inc

```diff
@@ -0,0 +1,6 @@
+payload
+~~~~~~~
+
+**type**: ``mixed`` **default**: ``null``
+
+This option can be used to attach arbitrary domain-specific data to a constraint.
```

Comment:
I agree with Wouter. We should keep it short, because I'm not sure this is useful for many people. But we can at least include his example to give people a hint. The error also appears to be available inside `FormError` instances. Though not guaranteed in the API, FormError::getCause() will return the Violation (see: https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Form/Extension/Validator/ViolationMapper/ViolationMapper.php#L132).